### PR TITLE
Add versioninig key & value checks

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-final-state/src/final_state.rs [hunk 1]
```diff
@@ -635,7 +635,16 @@ impl FinalState {
...
             } else if serialized_key.starts_with(MIP_STORE_PREFIX.as_bytes()) {
-                // TODO: check MIP_STORE_PREFIX
+                if !self
+                    .mip_store
+                    .is_key_value_valid(&serialized_key, &serialized_value)
+                {
+                    warn!("Wrong key/value for MIP Store");
+                    return Err(anyhow!(
+                        "Wrong key/value for MIP Store serialized_key: {:?}, serialized_value: {:?}", 
+                        serialized_key, serialized_value
```
Comment (bilboquet) [inline massa-final-state/src/final_state.rs:645]: I don't know how big `serialized_key` and `serialized_value` can be.
If they can be bigger than a "limit", maybe we want to use `.get(0..limit)` to return only their beginning

File: massa-final-state/src/final_state.rs [hunk 2]
```diff
@@ -648,6 +657,32 @@ impl FinalState {
...
+        {
+            if serialized_key.starts_with(MIP_STORE_PREFIX.as_bytes())
+                || serialized_key.starts_with(MIP_STORE_STATS_PREFIX.as_bytes())
+            {
+                if !self
+                    .mip_store
+                    .is_key_value_valid(&serialized_key, &serialized_value)
+                {
+                    warn!("Wrong key/value for MIP Store");
+                    return Err(anyhow!(
```
Comment (bilboquet) [inline massa-final-state/src/final_state.rs:671]: same as above

File: massa-final-state/src/final_state.rs [hunk 3]
```diff
@@ -648,6 +657,32 @@ impl FinalState {
...
+                        "Wrong key/value for MIP Store serialized_key: {:?}, serialized_value: {:?}",
+                        serialized_key, serialized_value
+                    ));
+                }
+            } else {
+                warn!(
+                    "Key/value does not correspond to any prefix: serialized_key: {:?}, serialized_value: {:?}",
+                    serialized_key, serialized_value
+                );
+                return Err(anyhow!(""));
```
Comment (bilboquet) [inline massa-final-state/src/final_state.rs:681]: why an empty message?

File: massa-versioning/src/versioning.rs [hunk 4]
```diff
@@ -1296,6 +1309,61 @@ impl MipStoreRaw {
         Ok(())
     }
 
+    // Final state
+    pub fn is_key_value_valid(&self, serialized_key: &[u8], serialized_value: &[u8]) -> bool {
+        self._is_key_value_valid(serialized_key, serialized_value)
+            .is_ok()
+    }
+
+    pub fn _is_key_value_valid(
```
Comment (bilboquet) [inline massa-versioning/src/versioning.rs:1318]: I'd vote for a different name `verify_key_value_validity` or maybe `check_key_value`
(I don't like much to have that two functions that differ only by a `_` in their name)

Comment (sydhds) [reply massa-versioning/src/versioning.rs:1318]: this is the very same function and _is_key_value_valid only exists for and if in unit tests we'd want to check for a particular error...

File: massa-final-state/src/final_state.rs (hunk 2 above)
Comment (sydhds) [reply massa-final-state/src/final_state.rs:673]: Done

File: massa-final-state/src/final_state.rs (hunk 3 above)
Comment (sydhds) [reply massa-final-state/src/final_state.rs]: Fixed

File: massa-final-state/src/final_state.rs (hunk 1 above)
Comment (bilboquet) [reply massa-final-state/src/final_state.rs:645]: fine

File: massa-versioning/src/versioning.rs (hunk 4 above)
Comment (bilboquet) [reply massa-versioning/src/versioning.rs:1318]: So shouldn't it be scoped for test non `pub`?
With `#[cfg(feature = "test-exports")]` the intent would be clearer.

File: massa-final-state/src/final_state.rs (hunk 3 above)
Comment (bilboquet) [reply massa-final-state/src/final_state.rs]: I'm still seeing `return Err(anyhow!(""));`
What am I missing?

**Review by bilboquet — APPROVED** [review]
Last opened discussion is not critical.
So I approve.

**Review by Leo-Besancon — APPROVED** [review]
LGTM!

**Pull request merged**